src/Schema/TableDiff.php

```
@@ -44,6 +44,17 @@ public function __construct(
         private readonly array $modifiedForeignKeys = [],
         private readonly array $droppedForeignKeys = [],
     ) {
+        if (count($this->modifiedForeignKeys) < 1) {
```

That's a convoluted way to check for `0`, or am I missing something?

---

Yes, it checks if the array is empty. In general, the value under `count()` could be any `Countable`, and its `count()` could be negative, so checking for `< 1` feels safer. But if it feels odd, we can agree on some other standard.

Personally for me, `=== 0` and `!== 0` looks too strict (both from the standpoints of requiring a strict type match and a strict value match). The value is guaranteed to be of type `int` but isn't guaranteed to be non-negative.

This is all purely aesthetic (or it's "defensive programming"?), so I'm fine with using whatever others are good with (except for `=== []` and `!== []`).

---

Why would you want to silently return for negative values? Semantically, I'd lean towards `=== []` , we want to detect if the value differs from the default value (regardless of what it is), right? I mean, we want to deprecate this parameter, don't we?

---

> Semantically, I'd lean towards `=== []` ,

This is a PHP-ism. See https://github.com/doctrine/dbal/pull/6621#discussion_r1855528698 for more details. Let's do `count($array) === 0` then.

---

Works for me